File: JavaScript/Programmers/Level 1/문자열 내림차순으로 배치하기.js
// 주어진 문자열을 내림차순 정렬

// function compare(a, b) { // sort 메소드에 들어갈 compare 함수
//     if (a > b) {
//         return -1;
//     } else if (a === b) {
//         return 0;
//     } else {
//         return 1;
//     }
// }

// function solution(s) {
//     // s를 배열로 변환 후 내림차순 정렬 후 다시 문자열로 join
//     return s.split("").sort((a, b) => compare(a, b)).join("");
// }

// compare 함수를 만들지 않고 삼항연산자 이용

// function solution(s) {
//     return s.split("").sort((a, b) => a > b ? -1 : 1).join("");
// }

// sort 메소드는 아스키코드값을 기준으로 오름차순 정렬하므로 오름차순 정렬 후 reverse 메소드를 사용하면 내림차순 정렬됨

function solution(s) {
    // s를 배열로 변환 후 오름차순 정렬 후 reverse하여 내림차순 정렬, 이후 다시 문자열로 join
    return s.split("").sort().reverse().join("");
}

File: Python/Programmers/Level 1/문자열 내림차순으로 배치하기.py
# def sort(s, reverse): # 정렬 구현, reverse에 따라 True면 내림차순, False면 오름차순
#     for i in range(len(s)):
#         for j in range(i, len(s)):
#             if reverse:
#                 if ord(s[i]) < ord(s[j]):
#                     s[i], s[j] = s[j], s[i]
#             else:
#                 if ord(s[i]) > ord(s[j]):
#                     s[i], s[j] = s[j], s[i]

#     return s

# def solution(s):
#     return "".join(sort(list(s), True)) # 정렬 후 문자열 반환

def solution(s):
    # sorted 함수를 사용하여 s를 정렬, 이때 reverse=True 값을 주어 내림차순 정렬 후 문자열로 변환
    return "".join(sorted(s, reverse=True))
